core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -390,8 +390,8 @@ class Partition(val topic: String,
       leaderReplica.highWatermark = newHighWatermark
       debug("High watermark for partition [%s,%d] updated to %s".format(topic, partitionId, newHighWatermark))
       true
-    } else {
-      debug("Skipping update high watermark since Old hw %s is larger than new hw %s for partition [%s,%d]. All leo's are %s"
+    } else  {
+      debug("Skipping update high watermark since new hw %s is not larger than old hw %s for partition [%s,%d]. All leo's are %s"
```

Comment:
Hmm, you swapped the order in the log message, but the parameters still have the old hw as the first parameter?